### PR TITLE
Fix docstring drawing of CHGate circuit element

### PR DESCRIPTION
**Pull request by KartikChinda**
- Sphinx was considering the circuit as a list item
- Fixed that to a normal circuit diagram.

### Summary
Fixing issue #7024
Issue link: https://github.com/Qiskit/qiskit-terra/issues/7024

### Details and comments
Since the documentation is created using Sphinx, it is considering the circuit diagram as a List element since the .. parsed-literal:: code comment isn't there. So I have added that to the CH gate, and that would seem to fix this. I tried it in an Online Sphinx editor and it worked.

### CODE REVIEW

**Review by kdk — APPROVED** [review]
Thanks @KartikChinda , this looks good!

File: qiskit/circuit/library/standard_gates/h.py [hunk 1]
```diff
@@ -109,6 +109,8 @@ class CHGate(ControlledGate):
     in the :math:`|1\rangle` state.
 
     **Circuit symbol:**
+
```
Comment (kdk) [inline qiskit/circuit/library/standard_gates/h.py:112]:
```suggestion

```
One lint failure due to extra whitespace on L112: https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=31402&view=logs&j=f0ab2c38-58fa-55ad-8e8b-d3984171f737&t=980625fc-4a64-50e4-7ca0-679c2a4bbda8&l=15 .

**Comment by KartikChinda** [comment]
Hi @kdk , thanks a lot. I've added the changes, since this is my first contribution, I'm not entirely sure but I hope this works now!

**Comment by 1ucian0** [comment]
You can build the docs and see the result of your changes with `tox -edocs`.

**Comment by jakelishman** [comment]
At the moment there's a mistake parsing the `.. parsed-literal::` directive, and it appears like this:

Screenshot 2021-09-20 at 16 36 05

You need a completely blank line between `**Circuit symbol**` and before `.. parsed-literal::` to have it work correctly.  Having whitespace in that line (like spaces or tabs) will cause pylint to moan (this was the problem before).

**Comment by 1ucian0** [comment]
@KartikChinda can you have look to @jakelishman [comment](https://github.com/Qiskit/qiskit-terra/pull/7034#issuecomment-923044217)? [Building the docs locally](https://github.com/Qiskit/qiskit-terra/pull/7034#issuecomment-922534539) to see result of your changes.

**Review by jakelishman — APPROVED** [review]
The generated docs are correct now, thanks!

**Comment by KartikChinda** [comment]
Hi @jakelishman ,  thanks a lot for your valuable help in fixing this issue! I had some work so I couldn't revert back sooner. And @1ucian0 , I did try building the docs but for some reason, it was showing an error in some package. But thanks for your help as well 😄!

**Comment by jakelishman** [comment]
@KartikChinda: that's strange that they wouldn't build for you.  If you can't build them in a completely clean environment (perhaps try `tox -r -edocs` - the `-r` rebuilds), you could open a bug report, since the docs should build on local machines as well.  We can help get to the bottom of that.

**Pull request merged**